File: CarSpecAngular/carspec/src/app/core/models/models/engineDto.ts
export class CreateEngineDto {
  engineName: string;
  fuelTypeId: number;
  numberOfCylinders: number;
  numberOfValves: number;
  displacement: number;
  maxPower: number;
  maxTorque: number;
  isTurbocharged: boolean;
  emissionStandard: string;

  constructor(
    engineName: string,
    fuelTypeId: number,
    numberOfCylinders: number,
    numberOfValves: number,
    displacement: number,
    maxPower: number,
    maxTorque: number,
    isTurbocharged: boolean,
    emissionStandard: string
  ) {
    this.engineName = engineName;
    this.fuelTypeId = fuelTypeId;
    this.numberOfCylinders = numberOfCylinders;
    this.numberOfValves = numberOfValves;
    this.displacement = displacement;
    this.maxPower = maxPower;
    this.maxTorque = maxTorque;
    this.isTurbocharged = isTurbocharged;
    this.emissionStandard = emissionStandard;
  }
}

export class EngineDto {
  engineId: number;
  engineName: string;
  fuelTypeId: number;
  numberOfCylinders: number;
  numberOfValves: number;
  displacement: number;
  maxPower: number;
  maxTorque: number;
  isTurbocharged: boolean;
  emissionStandard: string;

  constructor(
    engineId: number,
    engineName: string,
    fuelTypeId: number,
    numberOfCylinders: number,
    numberOfValves: number,
    displacement: number,
    maxPower: number,
    maxTorque: number,
    isTurbocharged: boolean,
    emissionStandard: string
  ) {
    this.engineId = engineId;
    this.engineName = engineName;
    this.fuelTypeId = fuelTypeId;
    this.numberOfCylinders = numberOfCylinders;
    this.numberOfValves = numberOfValves;
    this.displacement = displacement;
    this.maxPower = maxPower;
    this.maxTorque = maxTorque;
    this.isTurbocharged = isTurbocharged;
    this.emissionStandard = emissionStandard;
  }
}

File: CarSpecAPI/Data/Models/DataExtractionModel/EngineDto.cs
namespace CarSpecAPI.Data.Models.DataExtractionModel
{
    public sealed class EngineDto
    {
        public int IndexId { get; set; }
        public string EngineRef { get; set; } = null!;
        public string EngineName { get; set; } = null!;

        public int? NumberOfCylinders { get; set; }
        public int? NumberOfValves { get; set; }
        public decimal? Displacement { get; set; }

        public bool? IsTurbocharged { get; set; }

        public string? EmissionStandard { get; set; }
        public string? Aspiration { get; set; }
        public string? EngineType { get; set; }

        public List<EnginePerformanceDto> EnginePerformances { get; set; } = [];
    }
}
